NewHorizons/Builder/Props/ItemBuilder.cs

```diff
@@ -0,0 +1,185 @@
+                itemTypeName = itemName;
+            }
+
+            var itemType = GetOrCreateItemType(itemTypeName);
+
+            var item = go.GetAddComponent<NHItem>();
+            item._sector = sector;
+            item._interactable = info.interactRange > 0f;
+            item._interactRange = info.interactRange;
+            item._localDropOffset = info.dropOffset ?? Vector3.zero;
+            item._localDropNormal = info.dropNormal ?? Vector3.up;
+
+            item.DisplayName = itemName;
+            item.ItemType = itemType;
+            item.Droppable = info.droppable;
+            if (!string.IsNullOrEmpty(info.pickupAudio))
+            {
+                item.PickupAudio = AudioTypeHandler.GetAudioType(info.pickupAudio, mod);
+            }
+            if (!string.IsNullOrEmpty(info.dropAudio))
+            {
+                item.DropAudio = AudioTypeHandler.GetAudioType(info.dropAudio, mod);
+            }
+            if (!string.IsNullOrEmpty(info.socketAudio))
+            {
+                item.SocketAudio = AudioTypeHandler.GetAudioType(info.socketAudio, mod);
+            }
+            else
+            {
+                item.SocketAudio = item.DropAudio;
+            }
+            if (!string.IsNullOrEmpty(info.unsocketAudio))
+            {
+                item.UnsocketAudio = AudioTypeHandler.GetAudioType(info.unsocketAudio, mod);
+            }
+            else
+            {
+                item.UnsocketAudio = item.PickupAudio;
+            }
+            item.PickupCondition = info.pickupCondition;
+            item.ClearPickupConditionOnDrop = info.clearPickupConditionOnDrop;
+
+            Delay.FireInNUpdates(() =>
+            {
+                if (item != null && !string.IsNullOrEmpty(info.pathToInitialSocket))
+                {
+                    var socketGO = planetGO.transform.Find(info.pathToInitialSocket);
+                    if (socketGO != null)
+                    {
+                        var socket = socketGO.GetComponent<OWItemSocket>();
+                        if (socket != null)
+                        {
+                            if (socket.PlaceIntoSocket(item))
+                            {
+                                // Successfully socketed
+                            }
+                            else
+                            {
+                                NHLogger.LogError($"Could not insert item {itemName} into socket at path {socketGO}");
+                            }
+                        }
+                        else
+                        {
+                            NHLogger.LogError($"Could not find a socket to parent item {itemName} to at path {socketGO}");
+                        }
+                    }
+                    else
+                    {
+                        NHLogger.LogError($"Could not find a socket to parent item {itemName} to at path {socketGO}");
+                    }
+                }
+            }, 1);
```

Comment:
FireOnNextUpdate would be more clear here (heavy nitpick LOL)

Reply:
Fair enough, I copied and pasted from the socket code lol